kclvm/tools/src/LSP/src/request.rs

```diff
@@ -109,3 +113,58 @@ pub(crate) fn handle_completion(
     );
     Ok(res)
 }
+
+/// Called when a `Completion` request was received.
+pub(crate) fn handle_hover(
+    snapshot: LanguageServerSnapshot,
+    params: lsp_types::HoverParams,
+    sender: Sender<Task>,
+) -> anyhow::Result<Option<lsp_types::Hover>> {
+    let file = params
+        .text_document_position_params
+        .text_document
+        .uri
+        .path();
+
+    let (program, prog_scope, _) = parse_param_and_compile(
+        Param {
+            file: file.to_string(),
+        },
+        Some(snapshot.vfs),
+    )
+    .unwrap();
+    let kcl_pos = kcl_pos(file, params.text_document_position_params.position);
+    log_message(
+        format!(
+            "handle_hover {:?}",
+            params.text_document_position_params.position
+        ),
+        &sender,
+    )?;
+
+    let res = hover::hover(&program, &kcl_pos, &prog_scope);
+    Ok(res)
+}
+
+/// Called when a `GotoDefinition` request was received.
+pub(crate) fn handle_document_symbol(
+    snapshot: LanguageServerSnapshot,
+    params: lsp_types::DocumentSymbolParams,
+    sender: Sender<Task>,
+) -> anyhow::Result<Option<lsp_types::DocumentSymbolResponse>> {
+    let file = params.text_document.uri.path();
+
+    let (program, prog_scope, _) = parse_param_and_compile(
+        Param {
+            file: file.to_string(),
+        },
+        Some(snapshot.vfs),
+    )
+    .unwrap();
```

Comment:
s/parse_param_and_compile().unwrap()/parse_param_and_compile()?

Fix all issues like this.